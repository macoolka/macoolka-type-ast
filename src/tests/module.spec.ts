import { parseI18N } from '../'
import { writeFileSync } from 'fs'
import * as E from 'fp-ts/lib/Either'
import { pipe } from 'fp-ts/lib/pipeable'
import { format } from 'macoolka-prettier'
import * as path from 'path'
describe('module', () => {
    const container = path.join(__dirname, 'fixtures')
    it('basic', () => {
        pipe(
            parseI18N([{ container, folders: [], name: 'Basic.ts' }])({}),
            E.mapLeft(a=>{
                console.log(a)
                return a;
            }),
            E.map(a => {
                expect(a.length).toEqual(1)
                expect(a[0].path!.length > 0).toEqual(true)
                writeFileSync(__dirname+'/index.ts',format({parser:'json-stringify',content:JSON.stringify(a)}),'utf-8')
            }),
            E.mapLeft(a => {
                return a
            }),
            a => expect(E.isRight(a)).toEqual(true),
        )
    })
     it('locale', () => {
        const parseLocale = parseI18N([{ container, folders: [], name: 'Locale.ts' }])
        pipe(
            parseLocale({}),
            E.mapLeft(a=>{
                console.log(a)
                return a;
            }),
            a => {
                
                expect(E.isRight(a)).toEqual(true)
                return a
            },
            E.map(a => {
                writeFileSync(__dirname + '/locale.en.json', format({parser:'json-stringify',content:JSON.stringify(a)}), 'utf-8')
                expect(a).toMatchSnapshot()
                
            }),

        )
        pipe(
            parseLocale({ i18n: { locale: 'zh' } }),
            a => {
                expect(E.isRight(a)).toEqual(true)
                return a
            },
            E.map(a => {
                writeFileSync(__dirname + '/locale.zh.json', format({parser:'json-stringify',content:JSON.stringify(a)}), 'utf-8')
                expect(a).toMatchSnapshot()
                
            }),

        )

    }) 

})