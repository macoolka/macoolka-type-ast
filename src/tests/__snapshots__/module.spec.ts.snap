// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module locale 1`] = `
Array [
  Object {
    "_kind": "module",
    "classes": Array [
      Object {
        "_kind": "class",
        "deprecated": false,
        "description": Array [],
        "examples": Array [],
        "fields": Array [
          Object {
            "_kind": "field",
            "deprecated": false,
            "description": Array [
              "class field desc",
            ],
            "examples": Array [],
            "exclusiveCreate": false,
            "exclusiveLoad": false,
            "exclusiveSearch": false,
            "exclusiveUpdate": false,
            "exclusiveWhere": false,
            "id": false,
            "ignore": false,
            "name": "classfield",
            "order": true,
            "path": Array [],
            "readonly": false,
            "reason": Array [],
            "required": false,
            "signature": "classfield: string = ''",
            "since": "0.2.0",
            "type": "string",
            "unique": false,
          },
        ],
        "ignore": false,
        "implements": Array [],
        "isExported": true,
        "methods": Array [
          Object {
            "_kind": "method",
            "deprecated": false,
            "description": Array [
              "class method desc",
            ],
            "examples": Array [],
            "ignore": false,
            "name": "classMethod",
            "params": Array [],
            "path": Array [],
            "reason": Array [],
            "required": false,
            "returnVoid": false,
            "signature": "classMethod( { ... }",
            "since": "0.2.0",
            "type": "string",
          },
        ],
        "name": "Class",
        "path": Array [],
        "reason": Array [],
        "signature": "export class Class { ... }",
        "since": "0.2.0",
        "staticMethods": Array [
          Object {
            "_kind": "staticmethod",
            "deprecated": false,
            "description": Array [
              "class static method desc",
            ],
            "examples": Array [],
            "ignore": false,
            "name": "classStaticMethod",
            "params": Array [],
            "path": Array [],
            "reason": Array [],
            "required": false,
            "returnVoid": false,
            "signature": "static classStaticMethod( { ... }",
            "since": "0.2.0",
            "type": "string",
          },
        ],
      },
    ],
    "constants": Array [
      Object {
        "_kind": "constant",
        "deprecated": false,
        "description": Array [
          "constant desc",
        ],
        "examples": Array [],
        "ignore": false,
        "isExported": true,
        "name": "constanta",
        "path": Array [],
        "reason": Array [],
        "required": false,
        "signature": "export const constanta: 1 = ...",
        "since": "0.2.0",
        "type": "string",
      },
      Object {
        "_kind": "constant",
        "deprecated": false,
        "description": Array [
          "inner constant desc",
        ],
        "examples": Array [],
        "ignore": false,
        "isExported": true,
        "name": "exportvara",
        "path": Array [],
        "reason": Array [],
        "required": false,
        "signature": "export const exportvara: \\"\\" = ...",
        "since": "0.2.0",
        "type": "string",
      },
    ],
    "deprecated": false,
    "description": Array [
      "module desc",
    ],
    "examples": Array [],
    "exports": Array [
      Object {
        "_kind": "export",
        "deprecated": false,
        "description": Array [
          "export function desc",
        ],
        "examples": Array [],
        "ignore": false,
        "name": "exportmethod",
        "path": Array [],
        "reason": Array [],
        "signature": "() => any",
        "since": "0.2.0",
      },
      Object {
        "_kind": "export",
        "deprecated": false,
        "description": Array [
          "export constant desc",
        ],
        "examples": Array [],
        "ignore": false,
        "name": "exportvara",
        "path": Array [],
        "reason": Array [],
        "signature": "\\"\\"",
        "since": "0.2.0",
      },
    ],
    "file": true,
    "functions": Array [
      Object {
        "_kind": "function",
        "deprecated": false,
        "description": Array [
          "export inner function desc",
        ],
        "examples": Array [],
        "ignore": false,
        "isExported": true,
        "name": "exportmethod",
        "params": Array [],
        "path": Array [],
        "reason": Array [],
        "required": false,
        "returnVoid": false,
        "signature": "export const exportmethod = () => ...",
        "since": "0.2.0",
        "type": "string",
      },
      Object {
        "_kind": "function",
        "deprecated": false,
        "description": Array [
          "function desc",
        ],
        "examples": Array [],
        "ignore": false,
        "isExported": true,
        "name": "functiona",
        "params": Array [],
        "path": Array [],
        "reason": Array [],
        "required": false,
        "returnVoid": false,
        "signatures": Array [
          "export function functiona( { ... }",
        ],
        "since": "0.2.0",
        "type": "string",
      },
    ],
    "idUnique": false,
    "ignore": false,
    "interfaces": Array [
      Object {
        "_kind": "interface",
        "deprecated": false,
        "description": Array [
          "module desc",
        ],
        "examples": Array [],
        "fields": Array [],
        "ignore": false,
        "implements": Array [],
        "isExported": true,
        "methods": Array [],
        "name": "A",
        "path": Array [],
        "reason": Array [],
        "signature": "
interface A {

}
",
        "since": "0.2.0",
      },
      Object {
        "_kind": "interface",
        "deprecated": false,
        "description": Array [],
        "examples": Array [],
        "fields": Array [],
        "ignore": false,
        "implements": Array [],
        "isExported": true,
        "methods": Array [],
        "name": "B",
        "path": Array [],
        "reason": Array [],
        "signature": "
interface B {

}
",
        "since": "0.2.0",
      },
      Object {
        "_kind": "interface",
        "deprecated": false,
        "description": Array [
          "interface desc",
        ],
        "examples": Array [],
        "fields": Array [
          Object {
            "_kind": "field",
            "deprecated": false,
            "description": Array [
              "interface method var desc",
            ],
            "examples": Array [],
            "exclusiveCreate": false,
            "exclusiveLoad": false,
            "exclusiveSearch": false,
            "exclusiveUpdate": false,
            "exclusiveWhere": false,
            "id": false,
            "ignore": false,
            "name": "add",
            "order": true,
            "path": Array [],
            "readonly": false,
            "reason": Array [],
            "required": true,
            "signature": "add: () => void",
            "since": "0.2.0",
            "type": Object {
              "_kind": "type",
              "defaultEmptyArray": true,
              "isArray": false,
              "isArrayRequired": false,
              "maybe": false,
              "maybeArray": false,
              "value": "object",
            },
            "unique": false,
          },
          Object {
            "_kind": "field",
            "deprecated": false,
            "description": Array [
              "interface field desc",
            ],
            "examples": Array [],
            "exclusiveCreate": false,
            "exclusiveLoad": false,
            "exclusiveSearch": false,
            "exclusiveUpdate": false,
            "exclusiveWhere": false,
            "id": false,
            "ignore": false,
            "name": "string",
            "order": true,
            "path": Array [],
            "readonly": false,
            "reason": Array [],
            "required": true,
            "signature": "string: string",
            "since": "0.2.0",
            "type": Object {
              "_kind": "string",
              "defaultEmptyArray": true,
              "isArray": false,
              "isArrayRequired": false,
              "maybe": false,
              "maybeArray": false,
            },
            "unique": false,
          },
        ],
        "ignore": true,
        "implements": Array [
          "A",
          "B",
        ],
        "isExported": true,
        "methods": Array [
          Object {
            "_kind": "method",
            "deprecated": false,
            "description": Array [
              "interface method desc",
            ],
            "examples": Array [],
            "ignore": false,
            "name": "a",
            "params": Array [],
            "path": Array [],
            "reason": Array [],
            "required": true,
            "returnVoid": false,
            "signature": "a():void",
            "since": "0.2.0",
            "type": Object {
              "_kind": "type",
              "defaultEmptyArray": true,
              "isArray": false,
              "isArrayRequired": false,
              "maybe": false,
              "maybeArray": false,
              "value": "object",
            },
          },
        ],
        "name": "Interface",
        "path": Array [],
        "reason": Array [],
        "signature": "
interface Interface extends A,B{
  /**
   *interface method var desc
   */
add: () => void
  /**
   *interface field desc
   */
string: string  /**
   *interface method desc
   */
a():void
}
",
        "since": "0.2.0",
      },
    ],
    "name": "Locale",
    "path": Array [
      "",
      "home",
      "fastspeeed",
      "macoolka",
      "macoolka-type-ast",
      "src",
      "tests",
      "fixtures",
      "Locale.ts",
    ],
    "reason": Array [],
    "since": "0.2.0",
    "typealiases": Array [
      Object {
        "_kind": "typealias",
        "deprecated": false,
        "description": Array [
          "typeAlias desc",
        ],
        "examples": Array [],
        "fields": Array [],
        "ignore": false,
        "isExported": true,
        "methods": Array [],
        "name": "Type",
        "path": Array [],
        "reason": Array [],
        "required": false,
        "signature": "export type Type = {

 
}",
        "since": "0.2.0",
        "type": "string",
      },
    ],
  },
]
`;

exports[`module locale 2`] = `
Array [
  Object {
    "_kind": "module",
    "classes": Array [
      Object {
        "_kind": "class",
        "deprecated": false,
        "description": Array [],
        "examples": Array [],
        "fields": Array [
          Object {
            "_kind": "field",
            "deprecated": false,
            "description": Array [
              "类字段描述",
            ],
            "examples": Array [],
            "exclusiveCreate": false,
            "exclusiveLoad": false,
            "exclusiveSearch": false,
            "exclusiveUpdate": false,
            "exclusiveWhere": false,
            "id": false,
            "ignore": false,
            "name": "classfield",
            "order": true,
            "path": Array [],
            "readonly": false,
            "reason": Array [],
            "required": false,
            "signature": "classfield: string = ''",
            "since": "0.2.0",
            "type": "string",
            "unique": false,
          },
        ],
        "ignore": false,
        "implements": Array [],
        "isExported": true,
        "methods": Array [
          Object {
            "_kind": "method",
            "deprecated": false,
            "description": Array [
              "类方法描述",
            ],
            "examples": Array [],
            "ignore": false,
            "name": "classMethod",
            "params": Array [],
            "path": Array [],
            "reason": Array [],
            "required": false,
            "returnVoid": false,
            "signature": "classMethod( { ... }",
            "since": "0.2.0",
            "type": "string",
          },
        ],
        "name": "Class",
        "path": Array [],
        "reason": Array [],
        "signature": "export class Class { ... }",
        "since": "0.2.0",
        "staticMethods": Array [
          Object {
            "_kind": "staticmethod",
            "deprecated": false,
            "description": Array [
              "类静态方法描述",
            ],
            "examples": Array [],
            "ignore": false,
            "name": "classStaticMethod",
            "params": Array [],
            "path": Array [],
            "reason": Array [],
            "required": false,
            "returnVoid": false,
            "signature": "static classStaticMethod( { ... }",
            "since": "0.2.0",
            "type": "string",
          },
        ],
      },
    ],
    "constants": Array [
      Object {
        "_kind": "constant",
        "deprecated": false,
        "description": Array [
          "变量描述",
        ],
        "examples": Array [],
        "ignore": false,
        "isExported": true,
        "name": "constanta",
        "path": Array [],
        "reason": Array [],
        "required": false,
        "signature": "export const constanta: 1 = ...",
        "since": "0.2.0",
        "type": "string",
      },
      Object {
        "_kind": "constant",
        "deprecated": false,
        "description": Array [
          "变量描述",
        ],
        "examples": Array [],
        "ignore": false,
        "isExported": true,
        "name": "exportvara",
        "path": Array [],
        "reason": Array [],
        "required": false,
        "signature": "export const exportvara: \\"\\" = ...",
        "since": "0.2.0",
        "type": "string",
      },
    ],
    "deprecated": false,
    "description": Array [
      "模块描述",
    ],
    "examples": Array [],
    "exports": Array [
      Object {
        "_kind": "export",
        "deprecated": false,
        "description": Array [
          "导出函数描述",
        ],
        "examples": Array [],
        "ignore": false,
        "name": "exportmethod",
        "path": Array [],
        "reason": Array [],
        "signature": "() => any",
        "since": "0.2.0",
      },
      Object {
        "_kind": "export",
        "deprecated": false,
        "description": Array [
          "导出变量描述",
        ],
        "examples": Array [],
        "ignore": false,
        "name": "exportvara",
        "path": Array [],
        "reason": Array [],
        "signature": "\\"\\"",
        "since": "0.2.0",
      },
    ],
    "file": true,
    "functions": Array [
      Object {
        "_kind": "function",
        "deprecated": false,
        "description": Array [
          "函数描述",
        ],
        "examples": Array [],
        "ignore": false,
        "isExported": true,
        "name": "exportmethod",
        "params": Array [],
        "path": Array [],
        "reason": Array [],
        "required": false,
        "returnVoid": false,
        "signature": "export const exportmethod = () => ...",
        "since": "0.2.0",
        "type": "string",
      },
      Object {
        "_kind": "function",
        "deprecated": false,
        "description": Array [
          "函数描述",
        ],
        "examples": Array [],
        "ignore": false,
        "isExported": true,
        "name": "functiona",
        "params": Array [],
        "path": Array [],
        "reason": Array [],
        "required": false,
        "returnVoid": false,
        "signatures": Array [
          "export function functiona( { ... }",
        ],
        "since": "0.2.0",
        "type": "string",
      },
    ],
    "idUnique": false,
    "ignore": false,
    "interfaces": Array [
      Object {
        "_kind": "interface",
        "deprecated": false,
        "description": Array [
          "模块描述",
        ],
        "examples": Array [],
        "fields": Array [],
        "ignore": false,
        "implements": Array [],
        "isExported": true,
        "methods": Array [],
        "name": "A",
        "path": Array [],
        "reason": Array [],
        "signature": "
interface A {

}
",
        "since": "0.2.0",
      },
      Object {
        "_kind": "interface",
        "deprecated": false,
        "description": Array [],
        "examples": Array [],
        "fields": Array [],
        "ignore": false,
        "implements": Array [],
        "isExported": true,
        "methods": Array [],
        "name": "B",
        "path": Array [],
        "reason": Array [],
        "signature": "
interface B {

}
",
        "since": "0.2.0",
      },
      Object {
        "_kind": "interface",
        "deprecated": false,
        "description": Array [
          "接口描述",
        ],
        "examples": Array [],
        "fields": Array [
          Object {
            "_kind": "field",
            "deprecated": false,
            "description": Array [
              "接口方法变量描述",
            ],
            "examples": Array [],
            "exclusiveCreate": false,
            "exclusiveLoad": false,
            "exclusiveSearch": false,
            "exclusiveUpdate": false,
            "exclusiveWhere": false,
            "id": false,
            "ignore": false,
            "name": "add",
            "order": true,
            "path": Array [],
            "readonly": false,
            "reason": Array [],
            "required": true,
            "signature": "add: () => void",
            "since": "0.2.0",
            "type": Object {
              "_kind": "type",
              "defaultEmptyArray": true,
              "isArray": false,
              "isArrayRequired": false,
              "maybe": false,
              "maybeArray": false,
              "value": "object",
            },
            "unique": false,
          },
          Object {
            "_kind": "field",
            "deprecated": false,
            "description": Array [
              "接口字段描述",
            ],
            "examples": Array [],
            "exclusiveCreate": false,
            "exclusiveLoad": false,
            "exclusiveSearch": false,
            "exclusiveUpdate": false,
            "exclusiveWhere": false,
            "id": false,
            "ignore": false,
            "name": "string",
            "order": true,
            "path": Array [],
            "readonly": false,
            "reason": Array [],
            "required": true,
            "signature": "string: string",
            "since": "0.2.0",
            "type": Object {
              "_kind": "string",
              "defaultEmptyArray": true,
              "isArray": false,
              "isArrayRequired": false,
              "maybe": false,
              "maybeArray": false,
            },
            "unique": false,
          },
        ],
        "ignore": true,
        "implements": Array [
          "A",
          "B",
        ],
        "isExported": true,
        "methods": Array [
          Object {
            "_kind": "method",
            "deprecated": false,
            "description": Array [
              "接口方法描述",
            ],
            "examples": Array [],
            "ignore": false,
            "name": "a",
            "params": Array [],
            "path": Array [],
            "reason": Array [],
            "required": true,
            "returnVoid": false,
            "signature": "a():void",
            "since": "0.2.0",
            "type": Object {
              "_kind": "type",
              "defaultEmptyArray": true,
              "isArray": false,
              "isArrayRequired": false,
              "maybe": false,
              "maybeArray": false,
              "value": "object",
            },
          },
        ],
        "name": "Interface",
        "path": Array [],
        "reason": Array [],
        "signature": "
interface Interface extends A,B{
  /**
   *接口方法变量描述
   */
add: () => void
  /**
   *接口字段描述
   */
string: string  /**
   *接口方法描述
   */
a():void
}
",
        "since": "0.2.0",
      },
    ],
    "name": "Locale",
    "path": Array [
      "",
      "home",
      "fastspeeed",
      "macoolka",
      "macoolka-type-ast",
      "src",
      "tests",
      "fixtures",
      "Locale.ts",
    ],
    "reason": Array [],
    "since": "0.2.0",
    "typealiases": Array [
      Object {
        "_kind": "typealias",
        "deprecated": false,
        "description": Array [
          "类型描述",
        ],
        "examples": Array [],
        "fields": Array [],
        "ignore": false,
        "isExported": true,
        "methods": Array [],
        "name": "Type",
        "path": Array [],
        "reason": Array [],
        "required": false,
        "signature": "export type Type = {

 
}",
        "since": "0.2.0",
        "type": "string",
      },
    ],
  },
]
`;
